app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    category = benchmark_run.result.keys[0]
+
+    previous_benchmark_runs = BenchmarkRun.where('created_at < ?', benchmark_run.created_at).where(initiator_type: Commit, benchmark_type: benchmark_run.benchmark_type,
```

Comment:
> You can't use the id range to fetch the benchmark runs. You need to fetch the benchmark runs of the last 1000 commits ordered by the created at date of the commit.

I'll try to make it clearer.... 

1. Find the lastest 1000 commits **of the right repository** for the given benchmark_type and benchmark_result_type
2. Get the benchmark_run of those commits and form the result.